storybookjs/storybook
# Core: Replace `cpy` with `fs-extra` copy/copyFile

Issue: https://github.com/storybookjs/storybook/issues/15174

## What I did

Replaces the `cpy` library used by `@storybook/core-server` with the previously-installed `fs-extra` library. The `cpy` function was copying two things: the favicon and the contents of `prebuiltDir`.

For copying the favicon, I used the `copyFile` function, which needs to have the full paths of both the source and destination file, so I changed the destination to append the `basename` of the favicon to the output directory.

For copying the `prebuiltDir`, [the `copy` function](https://github.com/jprichardson/node-fs-extra/blob/master/docs/copy.md) in `fs-extra` will copy the contents of a directory into another directory recursively, which seems like is exactly what `cpy` was doing before. The only difference that I could see between the old use of `cpy` and `fs-extra` was that `fast-glob` [follows symlinks by default ](https://github.com/mrmlnc/fast-glob#followsymboliclinks) whereas `fs-extra` only seems to do so when `dereference` is `true`, so I added that option. (`fast-glob` is used by `globby` which is used by `cpy`)

## How to test

* `yarn test` completes successfully
* `yarn build-storybook` in the `examples/cra-ts` directory works correctly

Unfortunately, I couldn't figure out how to get the `prebuildDir` to show up - I added some logging to see if it ever got triggered, but none of the examples seemed to exercise it. I'm not sure how to enable that option to verify that it's working and would appreciate some help!


If your answer is yes to any of these, please make sure to include it in your PR.
